stat/pool.go

```diff
@@ -0,0 +1,84 @@
+package stat
+
+import (
+	"encoding/hex"
+	"sync"
+	"time"
+)
+
+// PoolStat is an external statistic for pool connections.
+type PoolStat struct {
+	errorThreshold uint32
+
+	mu       sync.RWMutex // protects nodeMonitor's map
+	monitors map[string]*nodeMonitor
+}
+
+// NewPoolStatistic is a constructor for [PoolStat].
+func NewPoolStatistic() *PoolStat {
+	return &PoolStat{
+		mu:       sync.RWMutex{},
```

Comment:
redundant

Reply:
The mutex protects the map inside `PoolStat`. Many callbacks for many nodes may come here, from different places

Reply:
i mean set struct field to zero, the mutex itself is required ofc. But maybe explicit is better than implicit